src/RuleHandlerContainer.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator\Rule\Handler\Container;
+
+use Psr\Container\ContainerInterface;
+use Yiisoft\Factory\Factory;
+use Yiisoft\Factory\NotFoundException;
+use Yiisoft\Validator\Exception\RuleHandlerInterfaceNotImplementedException;
+use Yiisoft\Validator\Exception\RuleHandlerNotFoundException;
+use Yiisoft\Validator\Rule\RuleHandlerInterface;
+use Yiisoft\Validator\RuleHandlerResolverInterface;
+
+final class RuleHandlerContainer implements RuleHandlerResolverInterface
+{
+    private array $handlers = [];
+    private Factory $factory;
+
+    public function __construct(ContainerInterface $container, array $definitions = [], bool $validate = true)
+    {
+        $this->factory = new Factory($container, $definitions, $validate);
```

Comment:
But why? We can use just only container.

Reply:
Two same-type rules may have different initialization data.

Reply:
> Two same-type rules may have different initialization data.

For example?

Reply:
Unique (table = post), unique (table = comment).

Reply:
> Unique (connection = db1), unique (connection = db2).

This case can be implenent via use container only. But this we want separate rule handlers from global container.

Reply:
> Unique (connection = db1), unique (connection = db2).

And what the rule and validator will look like?

Reply:
class UniqueRule extends ...
{
    private Connection $connection;

    public function __construct(Connection $connection)
    {
        $this->connection = $connection;
    }

    ...
}

Reply:
Couldn't it be a little bit more expressive? :)
I would like to see whole example